tests/controller-test.js

```diff
@@ -1,11 +1,530 @@
+    controllerMock.state.fullscreen = true; window.navigator.platform = "iPhone"; controllerMock.state.seeking = true;
+    Html5Skin.onPlayheadTimeChanged.call(controllerMock, 'customerUi', 5, 0, 7, null, 'child');
+    controllerMock.state.fullscreen = false; window.navigator.platform = 'Node'; controllerMock.state.seeking = false;
+
+    Html5Skin.onInitialPlay.call(controllerMock);
+
+    Html5Skin.onPlaying.call(controllerMock, 'customerUi', OO.VIDEO.MAIN);
+    Html5Skin.onPlaying.call(controllerMock, 'customerUi', OO.VIDEO.ADS);
+
+    Html5Skin.onPause.call(controllerMock, 'customerUi', OO.VIDEO.ADS, CONSTANTS.PAUSE_REASON.TRANSITION);
+    Html5Skin.onPaused.call(controllerMock, 'customerUi', 'video-id-123587dsjhkewu');
+    Html5Skin.onPaused.call(controllerMock, 'customerUi', OO.VIDEO.MAIN);
+    controllerMock.focusedElement = OO.VIDEO.MAIN;
+    controllerMock.state.screenToShow = CONSTANTS.SCREEN.DISCOVERY_SCREEN;
+    Html5Skin.onPaused.call(controllerMock, 'customerUi', OO.VIDEO.MAIN);
+
+    Html5Skin.onPlayed.call(controllerMock);
+    controllerMock.state.upNextInfo.delayedContentData = {clickedVideo: {embed_code: false}};
+    Html5Skin.onPlayed.call(controllerMock);
+    controllerMock.state.upNextInfo.delayedContentData = {clickedVideo: {embed_code: false}};
+    controllerMock.state.upNextInfo.delayedSetEmbedCodeEvent = null;
+    controllerMock.state.fullscreen = true; window.navigator.platform = "iPhone";
+    Html5Skin.onPlayed.call(controllerMock);
+    controllerMock.state.upNextInfo.delayedContentData = {clickedVideo: {embed_code: false}};
+    controllerMock.state.fullscreen = false; window.navigator.platform = "Node";
+
+    Html5Skin.onVcPlayed.call(controllerMock, 'customerUi', OO.VIDEO.MAIN);
+
+    Html5Skin.onSeeked.call(controllerMock, 'customerUi');
+    controllerMock.state.fullscreen = true; window.navigator.platform = "iPhone"; controllerMock.state.screenToShow = CONSTANTS.SCREEN.END_SCREEN;
+    Html5Skin.onSeeked.call(controllerMock, 'customerUi');
+    controllerMock.state.fullscreen = false; window.navigator.platform = "Node";
+
+    Html5Skin.onPlaybackReady.call(controllerMock, 'customerUi');
+
+    Html5Skin.onBuffering.call(controllerMock, 'customerUi');
+    controllerMock.state.isInitialPlay = true;
+    Html5Skin.onBuffering.call(controllerMock, 'customerUi');
+    controllerMock.state.isInitialPlay = false;
+
+    Html5Skin.onBuffered.call(controllerMock, 'customerUi');
+    controllerMock.state.buffering = true;
+    Html5Skin.onBuffered.call(controllerMock, 'customerUi');
+
+    Html5Skin.onReplay.call(controllerMock, 'customerUi');
+
+    Html5Skin.onAssetDimensionsReceived.call(controllerMock, 'customerUi', {});
+    controllerMock.skin.props.skinConfig.responsive.aspectRatio = 'auto';
+    Html5Skin.onAssetDimensionsReceived.call(controllerMock, 'customerUi', {videoId: OO.VIDEO.MAIN});
+
+    // test ad events
```

Comment:
This is harder to do because of the unique format of the controller and OO.plugin callback scope.

Low priority.

I don't want to spend anymore time on this.